docs/RLC-generation-Design.md

```diff
@@ -0,0 +1,57 @@
+# RLC generation Design
+
+## Basic Terms
+
+- **Cadl Files**: cloud service API definition written in [CADL](https://github.com/microsoft/cadl).
+- **Swagger Files**: REST api specification written in [OpenAPI](https://swagger.io/).
+- **CADL Program**: The model we get after CADL compiler has processed the cadl input.
+- **Modelerfour**: The code model we get after autorest core and modelerfour plugin has processed the swagger input.
+- **RLCModel**: The model we defined internally which contains all the necessary information we need to build the RLC file content.
+- **CADL compiler**: The module that compiles the cadl input and provides a set of APIs to CADL emitter to get the cadl program model.
+- **Autorest core & m4 plugin**: The module that takes swagger files as input and output the code model modelerfour.
+- **CADL TypeScript emitter**: This module transforms the CADL program into RLCModel.
+- **RLC common**: The common part that takes RLCModel as input and build the RLC file content. 
+- **Autorest TypeScript plugin**: This module transforms the Modelerfour into RLCModel.
+
+## TypeScript RLC generation Design Overview
+
+In the TypeScript RLC generator, we support generate RLC code from both cadl input and swagger input.
+
+```mermaid
+graph TD
+A[CADL Files] -->|CADL compiler| B(CADL Program)
+B -->|CADL TypeScript emitter| C{RLCModel}
+C --> |RLC common| D{RLC File Content}
+D --> |CADL TypeScript emitter| E(JS RLC Code)
+F(Swager Files) -->|Autorest core & m4 plugin| G(Modelerfour)
+G -->|Autotest TypeScript Plugin| C
```

Comment:
Not quite understand the path for 'RLC File Content' => 'JS RLC Code',  why we use different tool for the convert?

Reply:
Because what we get from rlc-common is just a file content string, we need to use either Autorest host to write the files or cadl emitter host to write the files.